petastorm/tests/test_predicates.py

```diff
@@ -124,3 +131,51 @@ def test_predicate_on_single_column(synthetic_dataset):
         actual = dict(row._asdict())
         assert actual['id2'] < 2
     assert counter == len(synthetic_dataset.data)
+
+
+def test_predicate_on_partitioned_dataset():
+    """
+    Generates a partitioned dataset and ensures that readers evaluate the type of the partition
+    column according to the type given in the Unischema.
+    """
+    TestSchema = Unischema('TestSchema', [
+        UnischemaField('id', np.int32, (), ScalarCodec(IntegerType()), False),
+        UnischemaField('test_field', np.int32, (), ScalarCodec(IntegerType()), False),
+    ])
+
+    def test_row_generator(x):
+        """Returns a single entry in the generated dataset."""
+        return {'id': x,
+                'test_field': x*x}
+
+    rowgroup_size_mb = 256
+    dataset_url = "file:///tmp/partitioned_test_dataset"
```

Comment:
This will make the test leave garbage directory once its done. We should use `tmpdir` fixture that is automatically provided for us by the framework. E.g. of usage: `test_simple_read_moved_dataset` test in test_end_to_end.

Reply:
Fixed.